lib/msf/ui/console/command_dispatcher/db/klist.rb

```diff
@@ -118,4 +134,51 @@ def cmd_klist(*args)
   def kerberos_ticket_storage
     @kerberos_ticket_storage ||= Msf::Exploit::Remote::Kerberos::Ticket::Storage::ReadWrite.new(framework: framework)
   end
+
+  private
+
+  def ticket_status(ticket)
+    if ticket.expired?
+      '>>expired<<'
+    elsif ticket.status.blank?
+      'active'
+    else
+      ticket.status
+    end
+  end
+
+  # Sets the status of the tickets
+  #
+  # @param [Symbol] mode The status (:activate or :deactivate) to apply to the ticket(s)
+  # @param [Array<StoredTicket>] tickets The tickets which statuses are to be updated
+  # @return [Integer] Number of tickets updated
+  def set_activation_status(mode, tickets)
+    result = []
+    if mode == :activate
+      result = kerberos_ticket_storage.activate_ccache(ids: tickets.map(&:id))
+    elsif mode == :deactivate
+      result = kerberos_ticket_storage.deactivate_ccache(ids: tickets.map(&:id))
+    end
+    result.size
+  end
+
+  def ticket_search(host_ranges, id_search)
+    ticket_results = []
+    each_host_range_chunk(host_ranges) do |host_search|
+      next if host_search && host_search.empty?
+
+      ticket_results += kerberos_ticket_storage.tickets(
+        workspace: framework.db.workspace,
+        host: host_search,
+        id: id_search # nil id returns all
+      )
+    end
+    host_ranges.each { |range| range.reset unless range.nil? } # Reset the Rex::Socket::RangeWalker so it can be re-used
+    ticket_results.sort_by(&:host_address)
```

Comment:
Does this logic need updated to be similar to:

```suggestion
    ticket_results.sort_by! { |ticket| [::IPAddr.new(ticket.host_address || "0").to_i, ticket.starttime.to_i] }
```

Reply:
I didn't change it to `sort_by` since it's the last line of the function so the new array gets returned anyway
but yep totally should have added in the proper ip sorting thanks for calling it out